### PR TITLE
feat: allow auth0 provider to specify connection/connection_scope config

### PR DESCRIPTION
**Pull request by imleozzito**
## Hey, I just made a Pull Request!

Now, Auth0 provider supports integration with social identity providers without creating friction or compromising the user experience. To check the available social connections, please visit [Auth0 Social Connections](https://marketplace.auth0.com/features/social-connections?_gl=1*6egh1p*rollup_ga*MTA1MTcyNDY1LjE2NjQ0NjA5Nzk.*rollup_ga_F1G3E656YZ*MTY2NDQ2NDEzNC4yLjEuMTY2NDQ2NDEzNy41Ny4wLjA.&_ga=2.201317486.1936071227.1664461101-105172465.1664460979).

In order to make this possible, `connection` and `connectionScope` configs were created (both optional), fully compatible with the `passport-auth` (added on [#13586](https://github.com/backstage/backstage/pull/13586)).

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR!
Some small tweaks otherwise looking good to me with my limited auth0 experience 👍 

@rugvip FYI

File: plugins/auth-backend/src/providers/auth0/provider.ts [hunk 1]
```diff
@@ -127,6 +133,10 @@ export class Auth0AuthProvider implements OAuthHandlers {
       scope: req.scope,
       state: encodeState(req.state),
       ...(this.audience ? { audience: this.audience } : {}),
+      ...(this.connection ? { connection: this.connection } : {}),
+      ...(this.connectionScope
```
Comment (Rugvip) [inline plugins/auth-backend/src/providers/auth0/provider.ts:137]: How do the `connectionScope` interact with the `req.scope`? passed from the frontend? Are they completely separate things, or are the situations where a user would expect the `req.scope` to be forwarded to `connectionScope`?

Comment (imleozzito) [reply plugins/auth-backend/src/providers/auth0/provider.ts:137]: @Rugvip Auth0 merges the pre-selected scopes with the ones provided in `connection_scope` by default. There is more information about it [here](https://community.auth0.com/t/pass-additional-scopes-permissions-to-a-social-connection/56103).

**Review by jhaals — APPROVED** [review]
Thank you 👏

**Comment by suuus** [comment]
Thank you for contributing during Hacktoberfest 🍂! You can claim the Backstage Hacktoberfest Holopin at https://bck.st/hacktoberfest-holopin 🙌🏻

**Pull request merged**